src/union_find.rs

```diff
@@ -1,6 +1,82 @@
+use core::panic;
+
 trait UnionFind {
```

Comment:
It would be useful to have some documentation on the generic contract exposed by this `trait` and by each of its methods.